src/main/java/org/jboss/arquillian/testcontainers/TestContainersObserver.java

```diff
@@ -21,35 +25,56 @@
 public class TestContainersObserver {
     @Inject
     @ClassScoped
-    protected InstanceProducer<GenericContainer<?>> containerWrapper;
+    protected InstanceProducer<TestContainerInstances> containersWrapper;
+    
+    protected ContainerRegistry registry;
```

Comment:
This should probably be `private`. Again, we shouldn't block this PR on this though. We _might_ simply be able to inject this too. We'd need to test that though :)

Reply:
Done